# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStoreParent.h
# Commit: 6a36e7a65173
# Full Hash: 6a36e7a65173113ea12080e630530655c190c0ab
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-13 04:28:43
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 10 - CookieStore API - CHIP implementation shared between Document and CookieStore, r=cookie-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D220288
# ==============================================================================

diff -r 2fbb65a885f6 -r 6a36e7a65173 dom/cookiestore/CookieStoreParent.h
--- a/dom/cookiestore/CookieStoreParent.h	Thu Sep 12 16:48:22 2024 +0000
+++ b/dom/cookiestore/CookieStoreParent.h	Thu Sep 12 16:48:23 2024 +0000
@@ -32,46 +32,55 @@
 
   mozilla::ipc::IPCResult RecvGetRequest(
       const nsString& aDomain, const OriginAttributes& aOriginAttributes,
-      const bool& aMatchName, const nsString& aName, const nsCString& aPath,
+      const Maybe<OriginAttributes>& aPartitionedOriginAttributes,
+      const bool& aThirdPartyContext, const bool& aPartitionForeign,
+      const bool& aUsingStorageAccess, const bool& aMatchName,
+      const nsString& aName, const nsCString& aPath,
       const bool& aOnlyFirstMatch, GetRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvSetRequest(
       const nsString& aDomain, const OriginAttributes& aOriginAttributes,
-      const nsString& aName, const nsString& aValue, const bool& aSession,
-      const int64_t& aExpires, const nsString& aPath, const int32_t& aSameSite,
-      const bool& aPartitioned, const nsID& aOperationID,
-      SetRequestResolver&& aResolver);
+      const bool& aThirdPartyContext, const bool& aPartitionForeign,
+      const bool& aUsingStorageAccess, const nsString& aName,
+      const nsString& aValue, const bool& aSession, const int64_t& aExpires,
+      const nsString& aPath, const int32_t& aSameSite, const bool& aPartitioned,
+      const nsID& aOperationID, SetRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvDeleteRequest(
       const nsString& aDomain, const OriginAttributes& aOriginAttributes,
-      const nsString& aName, const nsString& aPath, const bool& aPartitioned,
-      const nsID& aOperationID, DeleteRequestResolver&& aResolver);
+      const bool& aThirdPartyContext, const bool& aPartitionForeign,
+      const bool& aUsingStorageAccess, const nsString& aName,
+      const nsString& aPath, const bool& aPartitioned, const nsID& aOperationID,
+      DeleteRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvClose();
 
-  void GetRequestOnMainThread(const nsAString& aDomain,
-                              const OriginAttributes& aOriginAttributes,
-                              bool aMatchName, const nsAString& aName,
-                              const nsACString& aPath, bool aOnlyFirstMatch,
-                              nsTArray<CookieData>& aResults);
+  void GetRequestOnMainThread(
+      const nsAString& aDomain, const OriginAttributes& aOriginAttributes,
+      const Maybe<OriginAttributes>& aPartitionedOriginAttributes,
+      bool aThirdPartyContext, bool aPartitionForeign, bool aUsingStorageAccess,
+      bool aMatchName, const nsAString& aName, const nsACString& aPath,
+      bool aOnlyFirstMatch, nsTArray<CookieData>& aResults);
 
   // Returns true if a cookie notification has been generated while completing
   // the operation.
   bool SetRequestOnMainThread(ThreadsafeContentParentHandle* aParent,
                               const nsAString& aDomain,
                               const OriginAttributes& aOriginAttributes,
-                              const nsAString& aName, const nsAString& aValue,
-                              bool aSession, int64_t aExpires,
-                              const nsAString& aPath, int32_t aSameSite,
-                              bool aPartitioned, const nsID& aOperationID);
+                              bool aThirdPartyContext, bool aPartitionForeign,
+                              bool aUsingStorageAccess, const nsAString& aName,
+                              const nsAString& aValue, bool aSession,
+                              int64_t aExpires, const nsAString& aPath,
+                              int32_t aSameSite, bool aPartitioned,
+                              const nsID& aOperationID);
 
   // Returns true if a cookie notification has been generated while completing
   // the operation.
-  bool DeleteRequestOnMainThread(ThreadsafeContentParentHandle* aParent,
-                                 const nsAString& aDomain,
-                                 const OriginAttributes& aOriginAttributes,
-                                 const nsAString& aName, const nsAString& aPath,
-                                 bool aPartitioned, const nsID& aOperationID);
+  bool DeleteRequestOnMainThread(
+      ThreadsafeContentParentHandle* aParent, const nsAString& aDomain,
+      const OriginAttributes& aOriginAttributes, bool aThirdPartyContext,
+      bool aPartitionForeign, bool aUsingStorageAccess, const nsAString& aName,
+      const nsAString& aPath, bool aPartitioned, const nsID& aOperationID);
 
   CookieStoreNotificationWatcher* GetOrCreateNotificationWatcherOnMainThread(
       const OriginAttributes& aOriginAttributes);